meteor/server/publications/packageManager/expectedPackages/generate.ts

```diff
@@ -0,0 +1,250 @@
+import { PackageContainerOnPackage, Accessor, AccessorOnPackage } from '@sofie-automation/blueprints-integration'
+import { getContentVersionHash } from '@sofie-automation/corelib/dist/dataModel/ExpectedPackages'
+import { PeripheralDeviceId, ExpectedPackageId, PieceInstanceId } from '@sofie-automation/corelib/dist/dataModel/Ids'
+import { protectString, unprotectString } from '@sofie-automation/corelib/dist/protectedString'
+import {
+	PackageManagerExpectedPackage,
+	PackageManagerExpectedPackageBase,
+	PackageManagerExpectedPackageId,
+} from '@sofie-automation/shared-lib/dist/package-manager/publications'
+import deepExtend from 'deep-extend'
+import { ReadonlyDeep } from 'type-fest'
+import _ from 'underscore'
+import { getSideEffect } from '../../../../lib/collections/ExpectedPackages'
+import { Studio, StudioLight, StudioPackageContainer } from '../../../../lib/collections/Studios'
+import { clone, omit } from '../../../../lib/lib'
+import { CustomPublishCollection } from '../../../lib/customPublication'
+import { logger } from '../../../logging'
+import { ExpectedPackagesContentCache } from './contentCache'
+import type { StudioFields } from './publication'
+
+export async function regenerateForExpectedPackages(
```

Comment:
I think it would be good to document what this method is doing. Maybe the name could be more descriptive? regenerate what? Especially, since the "output" of this method, is actually the 4th argument.